core/src/apps/management/recovery_device/homescreen.py

```diff
@@ -21,32 +22,82 @@ async def recovery_homescreen() -> None:
     if not storage_recovery.is_in_progress():
         workflow.set_default(homescreen)
         return
-
-    await recovery_process()
+    elif (
+        storage_cache.get(storage_cache.APP_RECOVERY_REPEATED_BACKUP_UNLOCKED)
+        == b"\x01"
+    ):
+        await _continue_repeated_backup()
```

Comment:
i think those two branches need to be switched -- first look at the cached value, then at `is_in_progress`

(plus, `end_progress()` when finished the dry-run)

Reply:
> (plus, `end_progress()` when finished the dry-run)

Doesn't the `end_progress()` in `_finish_recovery_dry_run` count as such?